operatorcourier/api.py

```diff
@@ -41,42 +40,20 @@ def build_and_verify(source_dir=None, yamls=None, ui_validate_io=False,
     :raises OpCourierBadBundle: When the resulting bundle fails validation
     """
 
-    if source_dir is not None and yamls is not None:
-        logger.error("Both source_dir and yamls cannot be defined.")
-        raise TypeError(
-            "Both source_dir and yamls cannot be specified on function call.")
-
-    yaml_files = []
-
-    if source_dir is not None:
-        for filename in os.listdir(source_dir):
-            if filename.endswith(".yaml") or filename.endswith(".yml"):
-                with open(os.path.join(source_dir, filename)) as f:
-                    yaml_files.append(f.read())
-    elif yamls is not None:
-        yaml_files = yamls
-
-    bundle = BuildCmd().build_bundle(yaml_files)
-
-    valid, validation_results_dict = ValidateCmd(ui_validate_io).validate(bundle,
-                                                                          repository)
+    verification = Verification(source_dir, yamls, ui_validate_io, repository)
```

Comment:
Why are we removing the check to see if `source_dir` and `yamls` are both defined?

Reply:
Ah I see, it moved to the Verification class. But doesn't it seem like encapsulating the mechanical part should stay in the API method (creating folders, checking what type of input, etc. ?)

Reply:
@kevinrizza The check is moved to [here](https://github.com/jeremylinlin/operator-courier/blob/support-nested-directory-layout/operatorcourier/verify.py#L65). But all it does is just checking inputs. I think it makes sense to stay in that method. Even if we move it to the API method, we will still need this check in `get_manifests_info`, right?

Reply:
Not if you have separate methods under the hood that do the validation. Why not have separate constructors depending on the type of input you get? Abstracting the mechanics of what to do with the input and calling the right functions based on that seems like the only real job of the API method.